### ISSUE TITLE
Plugin generates unhygienic code

### ISSUE THREAD

**Issue by parsonsmatt** [well-typed/large-records#162]
Our custom prelude exports `error` as an alias for [`errorRequireCallStack`](https://hackage.haskell.org/package/require-callstack-0.2.0.0/docs/RequireCallStack.html#v:errorRequireCallStack), which causes the `large-record` plugin to fail with this compiler error message:

```
src/Handler/Perks/Types.hs:16:14-25: error: [GHC-64725]
    • Add RequireCallStack to your function context or use provideCallStack
    • In the expression:
        error
          "Pattern match failure in vectorToPerksSecrets: vector with invalid number o
f elements."
      In a case alternative:
          _ -> error
                 "Pattern match failure in vectorToPerksSecrets: vector with invalid n
umber of elements."
      In the expression:
        case Data.Record.Plugin.Runtime.anyArrayToList _x of
```

We can work-around this by hiding custom `error` and using `Prelude.error` but I think it'd be better if the plugin were hygienic about the terms it uses

**Comment by edsko** [comment]
Using `Exact` names in source plugins in a pain in the ass: you need to provide the location of where the symbol is _defined_; you cannot provide the location of a module that merely re-exports it. For this reason `large-records` comes with a `Data.Record.Plugin.Runtime` module that defines type aliases and term aliases for nearly everything we need. Unfortunately, we cannot define _class_ aliases; if you do

```haskell
type MyEq = Eq
```

then `ghc` will not accept

```haskell
instance MyEq T where ..
```

This means that for classes we must define the precise location of the class; for the `Prelude` names I wasn't doing this, and was just using whatever was in scope, to sidestep this issue. I have now pushed a fix to #164 that imports these names from `GHC.Classes` (`Eq` and `Ord`) and `GHC.Show` (`Show`) instead. Unfortunately, the former comes from `ghc-prim`, so the (small) price to pay is that that is now yet another dependency to add (for a total of three: `ghc-prim`, `large-records`, and `record-hasfield`). Once that PR passes CI I will make a release.

**Comment by edsko** [comment]
Released as https://hackage.haskell.org/package/large-records-0.4.2 .
